server/src/main/java/org/opensearch/search/profile/aggregation/AggregationProfiler.java

```diff
@@ -47,29 +48,185 @@
+                    breakdownTypeCount,
+                    breakdown.getOrDefault(breakdownTypeCount, 0L) + profileResult.getTimeBreakdown().get(breakdownTypeCount)
+                );
+            }
+
+            debug = profileResult.getDebugInfo();
+            children.addAll(profileResult.getProfiledChildren());
+        }
+        // nodeTime
+        long nodeTime = maxSliceNodeEndTime - minSliceNodeStartTime;
+        avgSliceNodeTime /= tree.size();
+
+        // Profiled breakdown time stats
+        for (AggregationTimingType breakdownTimingType : AggregationTimingType.values()) {
+            buildBreakdownMap(tree, breakdown, timeStatsMap, breakdownTimingType.toString());
+        }
+
+        // Profiled breakdown total time
+        for (AggregationTimingType breakdownTimingType : AggregationTimingType.values()) {
+            String breakdownType = breakdownTimingType.toString();
+            breakdown.put(breakdownType, maxSliceEndTimeMap.get(breakdownType) - minSliceStartTimeMap.get(breakdownType));
+        }
+
+        // Profiled breakdown count stats
+        for (String breakdownCountType : breakdownCountStatsTypes) {
+            buildBreakdownMap(tree, breakdown, countStatsMap, breakdownCountType);
+        }
+
+        // children
+        List<ProfileResult> reducedChildrenTree = new LinkedList<>();
+        if (!children.isEmpty()) {
+            Map<String, List<ProfileResult>> sliceLevelAggregationMap = getSliceLevelAggregationMap(children);
+            for (List<ProfileResult> profileResults : sliceLevelAggregationMap.values()) {
+                reducedChildrenTree.addAll(reduceProfileResultsTree(profileResults));
+            }
+        }
+
+        ProfileResult reducedResult = new ProfileResult(
+            type,
+            description,
+            breakdown,
+            debug,
+            nodeTime,
+            maxSliceNodeTime,
+            minSliceNodeTime,
+            avgSliceNodeTime,
+            reducedChildrenTree
+        );
+        List<ProfileResult> ret = new LinkedList<>();
+        ret.add(reducedResult);
+        return ret;
+    }
+
+    static void buildBreakdownMap(List<ProfileResult> tree, Map<String, Long> breakdown, Map<String, Long> statsMap, String breakdownType) {
+        String maxBreakdownType = "max_" + breakdownType;
+        String minBreakdownType = "min_" + breakdownType;
+        String avgBreakdownType = "avg_" + breakdownType;
+        breakdown.put(maxBreakdownType, statsMap.get(maxBreakdownType));
+        breakdown.put(minBreakdownType, statsMap.get(minBreakdownType));
+        breakdown.put(avgBreakdownType, statsMap.get(avgBreakdownType) / tree.size());
+    }
+
+    static void buildBreakdownStatsMap(Map<String, Long> statsMap, ProfileResult result, String breakdownType) {
+        String maxBreakdownType = "max_" + breakdownType;
+        String minBreakdownType = "min_" + breakdownType;
+        String avgBreakdownType = "avg_" + breakdownType;
```

Comment:
lets add static final constant variables like `private static final MAX_PREFIX = "max_"` for min/max/avg and reuse that in both the static utilities